app/models/infra_conversion_job.rb

```diff
@@ -131,7 +138,7 @@ def destination_vm
   end
 
   def target_vm
-    return @target_vm = source_vm if migration_phase == 'pre'
+    return @target_vm = source_vm if migration_phase == 'pre' || migration_task.canceling?
```

Comment:
What happens here if the migration phase is neither pre nor post nor canceling? Or is that simply not possible?

Reply:
`migration_phase` cannot be another value than `pre` or `post`.
The fact that the `task.options[:cancelation_status]` is set by the task. The cancelation logic will be moved to InfraConversionJob later with another PR.